docs/source/models/concept_models/vivarium_census_synthdata/concept_model.rst

```diff
@@ -2807,6 +2807,349 @@ have not been included. These are:
+    - Note that not all types of noise apply to all observers 
+  * - Income / Wages
+    - Taxes (both)
+    - 1%
+    - 1
+    - Numeric miswriting
+    - Note that wages and income are on separate tax forms and noise is applied to each separately 
+  * - Employer ID 
+    - Taxes (both)
+    - 1%
+    - 1
+    - Numeric miswriting
+    - 
+  * - Employer Name 
+    - Taxes (both)
+    - 1%
+    - 1
+    - OCR, typographic 
+    - 
+  * - Type of Tax Form  
+    - Taxes (both)
+    - 1%
+    - N/A
+    - Incorrect select 
+    - 
+  * - Type of SSA Event 
+    - SSA 
+    - 1%
+    - N/A
+    - Incorrect select 
+    - 
+  * - Date of SSA Event 
+    - SSA 
+    - 1%
+    - N/A
+    - Numeric miswriting, month and day swap 
+    - 
+
+The below section further describes types of noise including any code 
+available and information for implementation. **Software engineering team - please alert the research team if any of the below looks to be particularly challenging for further discussion.**
+
+**Converting from Noise Inputs to Noise Function Parameters for OCR, Phonetic, Typographic and Numeric Miswriting** 
+
+For simplicity for the end user, they will input two values for noise level: 
+
+#. Rows with Errors 
+#. Errors per Row 
+
+For types of noise others than those listed above, the only parameter is rows with error 
+and can be followed exactly. For noise types above, the functions we will use take different parameters: 
+
+#. Rows Eligible for Errors 
+#. Error Rate per Character OR per Token 
+
+These parameters, while related are different in key ways and therefore we need to 
+provide a way to convert between them. 
+
+Starting with #1 from both lists, we will calculate the number of rows eligible 
+for errors to approximate the user inputted rows with errors given. To calculate 
+this we can: 
+
+#. Compute the average length of strings in the column 
+#. Probability of at least 1 error = 1 - ((1 - (user defined errors per row / average string length))^(average string length))
+#. Rows eligible for errors = User defined rows with errors / probability of at least 1 error 
+
+Next, once a row is selected to receive noise, we define how much noise it should receive. 
+The default value here is 1 error per row, but can be changed. To calculate the error rate 
+per character/token, we can use this equation: 
+
+Error rate per Character/Token = (user defined errors per row)/(average string length)
+
+This approach has a number of assumptions and limitations: 
+
+- We assume that there are a similar number of tokens and characters in each string. This is approximately true for longer strings. While it is incorrect for short strings, we feel it is still reasonable overall. 
+- The user defined values for errors will not be exact, but rather approximated with this method. 
+- While the user provides an overall rate of errors per rows, in reality the number of expected errors per string is a function of the length of the string. 
```

Comment:
Since it's an average over all strings, I don't think the expected number of errors per string is a function of the length of the string. Rather, the conditional expectation of the number of errors, given that there is at least one error, is a function of the average number of tokens per string, which is unknown. The overall average number of errors per string also depends on the user-selected fraction of rows with errors.

Reply:
Fair point. I believe this was referring to the conditional expectation. We are not going to be able to do any of these calculations without an estimate for average number of tokens per string (see above where I have proposed how to get this). However, we'll also need to work out the _conditional_ expectation of tokens per string among those with at least one error, which (I think?) will require also the variance of the tokens-per-string distribution.